collectors/feed/collectors.py

```diff
@@ -0,0 +1,281 @@
+                skip_processing = True
+
+            if skip_processing:
+                # Prevent reprocessing next time
+                log.info("Skipping feed entry: %s", entry.link)
+                feed_models.append(
+                    SkippedFeedEntry(
+                        url=entry.link,
+                        skipped=True,
+                    )
+                )
+                continue
+
+            cves = find_cves(entry.title + text)
+            cwes = find_cwes(entry.title + text)
+            cvss = find_cvss(text)
+
+            blocklist, allowlist = check_keywords(entry.title + " " + text)
+            allowed = True
+            if blocklist and not allowlist:
+                log.info(
+                    "Skipping %s feed entry (%s) because of blocklist keywords (%s).",
+                    self.feed_name,
+                    entry.title,
+                    "; ".join(blocklist),
+                )
+                allowed = False
+
+            log.info(
+                "Continuing with %s feed entry (%s).",
+                self.feed_name,
+                entry.title,
+            )
+            feed_models.append(
+                FeedEntry(
+                    url=entry.link,
+                    cves=cves,
+                    title=entry.title,
+                    text=text.strip(),
+                    name=self.feed_name,
+                    source=self.source,
+                    components=self.components,
+                    cwe=cwes,
+                    cvss3=cvss,
+                    allowed=allowed,
+                    skipped=False,
+                )
+            )
+
+        log.info(
+            "Found %s new %s feed entries (unfiltered).",
+            len(feed_models),
+            self.feed_name,
+        )
+        count_allowed = len([f for f in feed_models if not f.skipped and f.allowed])
+        log.info(
+            "Found %s new %s feed entries (after filtering).",
+            count_allowed,
+            self.feed_name,
+        )
+
+        return feed_models
+
+
+def run_scraper(feed_data):
+    """Scrape a particular feed."""
+    feed_name = feed_data.get("name")
+    url = feed_data.get("url")
+    source = feed_data.get("source")
+
+    if not url or not feed_name or not source:
+        raise ValueError(
+            "Missing mandatory argument (url, feed name, source) in feed data: %s",
+            feed_data,
+        )
+
+    components = feed_data.get("components", [])
+
+    scraper = FeedScraper(feed_name, url, source, components)
+
+    return scraper.get_feed_data()
+
+
+def get_feed_data():
+    with open(FEEDS_FILE) as f:
+        feed_data = yaml.safe_load(f)
+    return feed_data
+
+
+def get_feed_entries():
+    """Concurrently scrape data from all feeds."""
+    log.info("Fetching new feeds")
+
+    feed_data = get_feed_data()
+
+    with ThreadPoolExecutor(MAX_COLLECTOR_WORKERS) as executor:
```

Comment:
In production the threading does not work for some reason. It seems that each pod runs on a single thread only. We encountered this when attempting to parallelize the flaw collector - it works locally but not in the production.

Also we should eventually think about what is the good way to do this. The collector framework prevents multiple concurrent runs of the same collector. We can however use Celery for the concurrency too so it is a question on which level we want to do it.

Reply:
There's currently no effective use of threading. This is a leftover from the original. I'll refactor this away. I prefer to gather feedback [like the one about Snippets](https://github.com/RedHatProductSecurity/osidb/pull/367#discussion_r1378609362) before actually doing the refactoring. See comment https://github.com/RedHatProductSecurity/osidb/pull/367#discussion_r1378022099 for additional explanation.

Reply:
Threading removed.